src/server/game/Chat/ChatCommandArgs.hpp

```diff
@@ -0,0 +1,494 @@
+/*
+ * Copyright (C) 2008-2018 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#ifndef TRINITY_CHATCOMMANDARGS_HPP
+#define TRINITY_CHATCOMMANDARGS_HPP
+
+#include "Optional.h"
+#include <boost/variant.hpp>
+#include <cstring>
+#include <string>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+class CommandArgs;
+
+namespace Trinity {
+namespace ChatCommandArgs {
+
+static const char COMMAND_DELIMITER = ' ';
+
+/***************** HELPERS *************************\
+|* These really aren't for outside use...          *|
+\***************************************************/
+inline size_t tokenize(char const*& end)
+{
+    size_t len = 0;
+    for (; *end && *end != COMMAND_DELIMITER; ++end, ++len);
+    for (; *end && *end == COMMAND_DELIMITER; ++end);
+    return len;
+}
+
+/************************** TAGS ********************************************************\
+|* Tags must implement the following methods:                                           *|
+|* - tryConsume: char const* -> char const*                                             *|
+|*   returns nullptr if no match, otherwise pointer to first character of next token    *|
+|* - get<T = value_type>: void -> value_type                                            *|
+|*   returns value                                                                      *|
+|* - is<T>: void -> bool                                                                *|
+|*   returns true iff value is present and of specified type                            *|
+|*                                                                                      *|
+\****************************************************************************************/
+class PlainInteger
+{
+    public:
+        template<typename T = int64>
+        T get() const
+        {
+            static_assert(std::is_integral_v<T>, "Attempt to extract PlainInteger using non-integral type");
```

Comment:
Recurring error in travis-ci:
```cpp
src/server/game/Chat/ChatCommandArgs.hpp:63:32:
 fatal error: no template named 'is_integral_v' in namespace 'std'; did you mean 'is_integral'?
```
is this intended, that you are planning to add code to define `is_integral_v` somewhere?